prow/metrics/metrics.go

```diff
@@ -26,23 +26,51 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/push"
 	"github.com/sirupsen/logrus"
+	"k8s.io/test-infra/prow/config"
 )
 
-// PushMetrics is meant to run in a goroutine and continuously push
+type configGetter interface {
+	Config() *config.Config
+}
+
+// Pusher is a struct that knows how to push metrics to prometheus
+type Pusher struct {
+	gatherer    gathererFunc
```

Comment:
Add a comment that this is helpful for testing purposes (I guess w/o having looked at your tests).